ambry-network/src/main/java/com.github.ambry.network/NetworkMetrics.java

```diff
@@ -121,13 +122,25 @@ public NetworkMetrics(MetricRegistry registry) {
     selectorNodeMetricMap = new HashMap<String, SelectorNodeMetric>();
   }
 
-  public void initializeSelectorMetricsIfRequired(final AtomicLong activeConnections) {
-    selectorActiveConnections = new Gauge<Long>() {
+  /**
+   * Initializes a few network metrics for the selector
+   * @param activeConnections count of current active connections
+   * @param numConnectionsPendingHandhsake Count of connections that are awaiting handshake completion
+   */
+  public void initializeSelectorMetricsIfRequired(final AtomicLong activeConnections,
+      final AtomicLong numConnectionsPendingHandhsake) {
+    numActiveConnections = new Gauge<Long>() {
       @Override
       public Long getValue() {
         return activeConnections.get();
       }
     };
+    this.numConnectionsPendingHandhsake = new Gauge<Long>() {
```

Comment:
looks like you named the other arg as `activeConnections` so that the `this` is avoided. Could you do the same here? (or name the other also as `numActiveConnections`). This is minor but I would like some consistency.